File: backend/src/main/java/tp/desarrollo/clases/Habitacion.java
package tp.desarrollo.clases;

import java.util.List;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToMany;
import lombok.Data;

@Entity
@Data
public class Habitacion {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long numeroHabitacion;
    private String tipo;
    private int cantidadHuespedes;
    private int cantidadCamaI;
    private int cantidadCamaD;
    private int cantidadCamaKS;

    @OneToMany (cascade = CascadeType.ALL)
    @JoinColumn(name = "habitacion_id")
    private List<EstadoHabitacion> historiaEstados;

}

File: frontend/interfaces/Habitacion.ts
import { EstadoHabitacion } from './EstadoHabitacion';

export interface Habitacion {
  numeroHabitacion: number; 
  tipo: string;
  cantidadHuespedes: number;
  cantidadCamaI: number;
  cantidadCamaD: number;
  cantidadCamaKS: number;
  historiaEstados: EstadoHabitacion[]; 
}